client/reducer/fish.js

```diff
@@ -3,6 +3,9 @@ export default function fishReducer (state = [], action) {
     case 'GET_FISH_SUCCESS':
       return action.fish
 
+    case 'GET_FISH_DETAIL':
+      return action.fishDetail
```

Comment:
This could be a problem. Here you're replacing the entire fish array (`state = []`) with a single `fishDetail` object. It'd be better to have a separate `details` reducer file, or else find the specific fish inside the array and add a details property to that.